VisualStudio/src/CookingPatches.cs

```diff
@@ -10,19 +10,26 @@ internal class CookingPotItem_DoSpecialActionFromInspectMode
         internal static bool Prefix(CookingPotItem __instance)
         {
             float waterAmount = __instance.m_LitersWaterBeingBoiled;
-            if (__instance.GetCookingState() == CookingPotItem.CookingState.Ready && waterAmount > 0) //patch does not apply while cooking
+            if (waterAmount <= 0) //only applies with water 
+            {
+                return true;
+            }
+            bool is_ready = __instance.GetCookingState() == CookingPotItem.CookingState.Ready;
+            bool is_not_ready_and_no_fire = __instance.GetCookingState() == CookingPotItem.CookingState.Cooking && !__instance.AttachedFireIsBurning();
+            if (is_ready || is_not_ready_and_no_fire) //patch applies if ready or if still cooking but no fire.
```

Comment:
All of this is to work when the water is non-potable water (only melted but boiled). Basically, only applies on the two possible scenarios: (is_ready, is not ready but no fire is burning [you can drink this])